src/features/dappLogin/DappLoginScreen.tsx

```diff
@@ -67,23 +66,18 @@ const DappLoginScreen = () => {
     }
   }, [pairClient, params.callback, params.uri])
 
-  const goBack = useCallback(() => {
+  const goBack = useCallback(async () => {
+    await disconnect()
     if (navigation.canGoBack()) {
       navigation.goBack()
     } else {
       navigation.replace('AccountsScreen')
     }
-  }, [navigation])
-
-  useDisappear(() => {
```

Comment:
This was triggering when the lock screen popped up causing the screen to just auto dismiss. I added the disconnect to goBack() instead